json_yaml_files/prepare_files.py

```diff
@@ -0,0 +1,178 @@
+                'fsip': 'MR structural (FISP)',
+                't2w': 'MR structural (T2)',
+                'pd_t2': 'MR structural (PD, T2)',
+                'b0_map': 'MR structural (B0 map)',
+                'b1_map': 'MR structural (B1 map)',
+                'flash': 'MR structural (FLASH)',
+                'mp2rage': 'MR structural (MP2RAGE)',
+                'tse': 'MR structural (TSE)',
+                't1w_t2w': 'MR structural (T1, T2)',
+                'mpnrage': 'MR structural (MPnRAGE)'
+            },
+            'pet': {
+                'pet': 'PET'
+            }
+        }
+        self.image_modality = {
+            'pet': 'PET',
+            'anat': 'MRI',
+            'func': 'MRI',
+            'dwi': 'MRI',
+            'fmap': 'MRI'
+        }
+
+    def extract_file_name_components(self, file_name):
+        match = re.match(self.pattern, file_name)
+        if match:
+            A = match.group('A')
+            X = match.group('X')
+            Y = match.group('Y')
+            Z = match.group('Z')
+            return A, X, Y, Z
+        else:
+            return None
+
+    def prepare_json_contents(self, file_name):
+        # json_contents = {file: file for file in self.top_level_files}
+        json_contents = {}
+        _, X, Y, _ = self.extract_file_name_components(file_name)
+        files = self.layout.get(
+            scope=X if X != 'inputs' else 'raw', datatype=Y, return_type='file')
+        for file_path in files:
+            sub_index = file_path.find('derivatives')
+            if sub_index == -1:
+                sub_index = file_path.find('sub-')
+            if sub_index != -1:
+                sub_path = file_path[sub_index:]
+                entities = self.layout.parse_file_entities(sub_path)
+                subject = entities.get('subject')
+                session = entities.get('session')
+                if subject:
+                    sub_path = sub_path.replace(subject, '{SUBJECT}')
+                if session:
+                    sub_path = sub_path.replace(session, '{SESSION}')
+                sub_path = re.sub(r'-\d', '{#}', sub_path)
+                json_contents[sub_path] = sub_path
+        finalPath = '/'.join(self.path.split('/')[:-2]) + '/prepared_jsons'
+        print(finalPath)
+        Path(finalPath).mkdir(parents=True, exist_ok=True)
+        with open(f'{finalPath}/'+file_name, 'w') as f:
+            json.dump(json_contents, f)
+
+    def fetch_scan_type(self, Y, Z):
+        scan_type = ''
+        if Z.lower() in self.scan_type[Y]:
+            scan_type = self.scan_type[Y][Z.lower()]
+        elif '_' in Z:
+            first = Z.split('_')[0]
+            if first.lower() in self.scan_type[Y]:
+                scan_type = self.scan_type[Y][first.lower()]
+        if Y in self.scan_type[Y]:
+            scan_type = self.scan_type[Y][Y]
+        return scan_type
+
+    def prepare_yaml_contents(self, file_name):
```

Comment:
Should this be indicative of "only for use with image03"? Or perhaps a `data_structure` argument to generalize it and then it can be tested against for NDA data structures besides `image03` in the future?

```suggestion
    def prepare_yaml_contents(self, file_name, data_structure='image03'):
```

Reply:
Or just refer back to `self.A`, I almost forgot about that object...

Reply:
yes. Should I take A value from the user using the arguments?

Reply:
I think that's a great idea! But make sure it's only got `choices` that are supported by the underlying code so far, with the `choices` option to `parser.add_argument()`.